ledger/complete/wal/checkpoint_v6_leaf_reader.go

```diff
@@ -106,4 +108,6 @@ func readCheckpointSubTrieLeafNodes(leafNodesCh chan<- LeafNodeResult, dir strin
 			Err:      err,
 		}
 	}
+
+	return err
```

Comment:
Since `readCheckpointSubTrieLeafNodes` returns error now, this proceeding block to push wrapped error to channel can be removed.  

```
	if err != nil {
		leafNodesCh <- LeafNodeResult{
			LeafNode: nil,
			Err:      err,
		}
	}
```

Also, `LeafNodeResult` isn't needed because errors are returned by functions,  
so `chan<- LeafNodeResult` can be `chan<- LeafNode` to simplify consumer
and save a little bit of memory.

Reply:
Great idea!